modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/manager/ExtensionActivator.java

```diff
@@ -80,7 +80,7 @@ private void registerEnumTransformers(ExtensionModel extensionModel) {
           final Class<Enum> enumClass = (Class<Enum>) type;
           if (enumTypes.add(enumClass)) {
             try {
-              StringToEnum stringToEnum = new StringToEnum(enumClass);
+              StringToEnum stringToEnum = new StringToEnum(enumClass, extensionModel.getName());
               registerObject(muleContext, getName(stringToEnum), stringToEnum, Transformer.class);
```

Comment:
can you add an integration test which verifies the case in which two different extensions register enums with the exact same FQN ?